Tests/Provider/Fixtures/TestBootCommand.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Knp\Tests\Provider\Fixtures;
+
+use Knp\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class TestBootCommand extends Command
+{
+    protected function configure()
+    {
+        parent::configure();
+        $this->setName('test:boot')
+            ->setDescription('Test boot command.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        /** @var TestBootApplication $silex */
+        $silex = $this->getSilexApplication();
+
+        $output->write($silex->isBooted() ? 'Booted' : 'Not booted');
+    }
```

Comment:
I'm not sure I get what you mean. The actual assertion is in the test, and that's where it should be in my opinion.

Now, considering that the command is only used for this test, something like this would also work here:
```php
<?php
    if (!$silex->isBooted()) {
        throw new \RuntimeException('The Silex application must boot before the command execution.');
    }
    $output->write('Booted');
```

But to me it doesn't change the semantics of the test: the command does _A_ or _B_ and we assert that it does _A_ in the test, as it's what we expect. 

Now, I could change it if you think that would be better.